message User {
	required int32 age = 1;
	required string department = 2;
	required string degree = 3;
	required string degreeYear = 4;
}

message MobileOS {
	required string os = 1;
	required string os_version = 2;
}

message MobileDevice {
	required string brand = 1;
	required string model = 2;
	required bool audio_support = 3;
	required bool video_support = 4;
}

message NetworkDetails {
	optional string ip = 1;
	required double bandwidth = 2;	
}

message RegisterUser {
	required string username = 1;
	required User user = 2;
	required MobileOS mobile_os = 3;
	required MobileDevice mobile_device = 4;
	required RSAKey public_key = 5;
}

message RSAKey {
    required string mod = 1;
    required string exp = 2;
}

message RegisterUserResponse {
	required int64 user_id = 1;
}

message ExchangePublicKey {
	required int64 user_id = 1;
	required string publicKeyClient = 2;
}

message ExchangePublicKeyResponse {
	required string publicKeyService = 1;
}

message SendQuery {
	required int64 user_id = 1;
	required string	query = 2;
}

message SendQueryResponse {
	repeated QueryResult query_result = 1;
}

message QueryResult {
	required string title = 1;
	required string url = 2;
}
